internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/WriteComputeWithActiveActiveReplicationTest.java

```diff
@@ -0,0 +1,1184 @@
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(0)), storeName, key2, timestampedOp);
+    verifyFLMRecord(storeName, dc1RouterUrl, key2, regularFieldDefault, intArrayFieldDefault, expectedMapFieldValue);
+    verifyFLMRecord(storeName, dc0RouterUrl, key2, regularFieldDefault, intArrayFieldDefault, expectedMapFieldValue);
+
+    expectedMapFieldValue.remove("OneZero");
+    expectedMapFieldValue.remove("TwoZero");
+
+    // PUT
+    // Should keep elements with higher timestamps only
+    regularFieldValue = "Key2F1";
+    arrayFieldValue = Collections.emptyList();
+    mapFieldValue = new HashMap<>();
+    mapFieldValue.put("one", 1);
+    mapFieldValue.put("two", 2);
+    mapFieldValue.put("three", 3);
+    GenericRecord val2 = new GenericData.Record(valueSchemaV1);
+    val2.put(REGULAR_FIELD, regularFieldValue);
+    val2.put(INT_ARRAY_FIELD, arrayFieldValue);
+    val2.put(MAP_FIELD, mapFieldValue);
+    timestampedOp = new VeniceObjectWithTimestamp(val2, 2);
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(0)), storeName, key2, timestampedOp);
+    expectedMapFieldValue.putAll(mapFieldValue);
+    verifyFLMRecord(storeName, dc0RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+    verifyFLMRecord(storeName, dc1RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+
+    // AddToMap: Update should be applied since the lowest timestamp of elements in the map(T2) <= timestamp of WC op
+    // (t3)
+    ub = new UpdateBuilderImpl(wcSchemaV1);
+    mapFieldValue = new HashMap<>();
+    mapFieldValue.put("four", 4);
+    ub.setEntriesToAddToMapField(MAP_FIELD, mapFieldValue);
+    timestampedOp = new VeniceObjectWithTimestamp(ub.build(), 3);
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(1)), storeName, key2, timestampedOp);
+    expectedMapFieldValue.putAll(mapFieldValue);
+    verifyFLMRecord(storeName, dc1RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+    verifyFLMRecord(storeName, dc0RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+
+    ub = new UpdateBuilderImpl(wcSchemaV1);
+    mapFieldValue = new HashMap<>();
+    mapFieldValue.put("five", 5);
+    ub.setEntriesToAddToMapField(MAP_FIELD, mapFieldValue);
+    timestampedOp = new VeniceObjectWithTimestamp(ub.build(), 5);
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(0)), storeName, key2, timestampedOp);
+    expectedMapFieldValue.putAll(mapFieldValue);
+    verifyFLMRecord(storeName, dc1RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+    verifyFLMRecord(storeName, dc0RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+
+    // AddToMap: The following update is ignored.
+    // todo: We need to have tie-breaking mechanism based on value when timestamps of modifications are the same
+    /*
+    ub = new UpdateBuilderImpl(wcSchemaV1);
```

Comment:
Is this part still useful? I believe tie breaking also looks at value hashcode?

Reply:
For map collection merging , I believe it is using colo ID to distinguish updates from different COLO with the same timestamp. For the update from the same colo, I beleive the order is guaranteed.
Checkout SortBasedCollectionFieldOpHandler last line:       return currCollectionFieldRmd.getTopLevelColoID() > incomingRequestColoID;

Reply:
As we discussed offline, we can also try to split the work to multiple stage. Feel free to let me know!